File: src.16/Domain/Entities/Accounting/Consignment.cs
using System;
using System.Collections.Generic;
using System.Data.Entity;

using Luxena.Domain;


namespace Luxena.Travel.Domain
{

	[RU("Накладная", "Накладные")]
	[SupervisorPrivileges]
	public partial class Consignment : Entity2
	{
		[EntityName, Patterns.Number, Length(12)]
		public string Number { get; set; }

		[EntityDate, Patterns.IssueDate]
		public DateTimeOffset IssueDate { get; set; }

		[RU("Отпущено")]
		protected Party _Supplier;

		[RU("Получено")]
		protected Party _Acquirer;

		[ReadOnly]
		public virtual Order Order
		{
			get { return OrderItems.OneAs(a => a.Order); }
		}

		
		[RU("Сумма с НДС")]
		public Money GrandTotal { get; set; }

		[Patterns.Vat]
		public Money Vat { get; set; }

		[RU("Сумма без НДС")]
		public Money Total
		{
			get { return GrandTotal - Vat; }
		}

		[Patterns.Discount]
		public Money Discount { get; set; }

		[RU("Всего отпущено")]
		public string TotalSupplied { get; set; }


		public virtual ICollection<OrderItem> OrderItems { get; set; }

		public virtual ICollection<IssuedConsignment> IssuedConsignments { get; set; }


		static partial void Config_(Domain.EntityConfiguration<Consignment> entity)
		{
			entity.Association(a => a.Supplier);//, a => a.Consignments_Supplier);
			entity.Association(a => a.Acquirer);//, a => a.Consignments_Acquirer);
		}

	}


	partial class Domain
	{
		public DbSet<Consignment> Consignments { get; set; }
	}

}

File: src.16/Web/app/Accounting/Consignment.ts
module Luxena.Views
{
	registerEntityControllers(sd.Consignment, se => ({

		list: [
			se.IssueDate,
			se.Number,
			se.Supplier,
			se.Acquirer,
			se.TotalSupplied,
			se.GrandTotal,
		],

		view: {
			"fields1": [
				se.IssueDate,
				se.Number,
				se.Supplier,
				se.Acquirer,
				se.TotalSupplied,
			],
			"fields2": [
				se.Discount,
				se.Total,
				se.Vat,
				se.GrandTotal,
			],
		},

		viewScope: ctrl => ({
			tabs: [
				se.OrderItems.toTab(ctrl, a => [a.Order, a.Position, a.Product, a.Text, a.GrandTotal, ]),
				se.IssuedConsignments.toTab(ctrl, a => [a.TimeStamp, a.Number, a.IssuedBy, ]),
			]
		}),


		edit: {
			"fields": [
				se.Number,
				se.IssueDate,
				se.Supplier,
				se.Acquirer,
				se.TotalSupplied,
			],
		},

	}));


	registerEntityControllers(sd.IssuedConsignment, se => [
		se.Consignment, se.TimeStamp, se.Number, se.IssuedBy,
	]);
}